src/x/crossCorrelation.js

```diff
@@ -0,0 +1,43 @@
+import { multiply } from './multiply';
+
+/**
+ * Calculates the cross-correlation between 2 vectors
+ * @param {Array} [A] - the array that will be fixed
+ * @param {Array} [B]
+ * @return {Array}
+ */
+// let multiply = SP.X.multiply;
+
+export function crossCorrelation(fun1, fun2, options = {}) {
+  let { tau, lag } = options;
+  if (tau === undefined) {
+    tau = 1;
+  }
+  if (lag === undefined) {
+    lag = fun1.length;
+  }
+  if (!fun2) {
+    fun2 = fun1;
+  }
+  if (fun2.tau) {
+    tau = fun2.tau;
+    lag = fun2.lag;
+    fun2 = fun1;
+  }
+  let result = [];
+  if (fun1.length === fun2.length) {
+    let n = fun2.length;
+    let mlag = n - lag;
+    let g = new Array(2 * n).fill(0);
```

Comment:
Use float64 array. They are initialised to 0 and they are faster: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Float64Array